src/els_code_navigation.erl

```diff
@@ -106,4 +101,6 @@ add_include_uri(#{ id := String }, Acc) ->
 beginning() ->
   #{range => #{from => {1, 1}, to => {1, 1}}}.
 
-%% TODO: Handle multiple header files with the same name?
+-spec filename_to_atom(string()) -> atom().
+filename_to_atom(FileName) ->
+  list_to_atom(filename:basename(FileName, filename:extension(FileName))).
```

Comment:
Yes, dinamically creating atoms. But we can't always ensure they always exist (and therefore use the `to_existing_atom` variant here. Hopefully, this is not going to be a problem...